packages/amplify-category-function/src/provider-utils/awscloudformation/utils/consolidateDependsOn.ts

```diff
@@ -0,0 +1,28 @@
+import { $TSMeta } from 'amplify-cli-core';
+import { FunctionDependency } from 'amplify-function-plugin-interface';
+import { lambdaLayerSetting, resourceAccessSetting, ServiceName } from './constants';
+
+export const consolidateDependsOnForLambda = (
+  projectMeta: $TSMeta,
+  currentDependsOn: FunctionDependency[],
+  lambdaToUpdate: string,
+  selectedSettings: string[],
+): FunctionDependency[] => {
+  let updatedDependsOn: FunctionDependency[];
+  const prevFunctionParametersDependsOn: FunctionDependency[] = projectMeta?.['function']?.[`${lambdaToUpdate}`]?.['dependsOn'] ?? [];
+  if (selectedSettings.includes(resourceAccessSetting)) {
+    // insert old lambdaLayer dependsOn if present
+    const prevLayersDependsOn: FunctionDependency[] = prevFunctionParametersDependsOn.filter(
+      resource => (projectMeta?.['function']?.[`${resource.resourceName}`]?.['service'] ?? []) === ServiceName.LambdaLayer,
```

Comment:
nit: here and LN 22 simplifies to

```ts
resource => (projectMeta?.function?.[`${resource.resourceName}`]?.service) === ServiceName.LambdaLayer,
```

because the default value is not needed, and dot notation is preferred when possible.